cmd/geth/main.go

```diff
@@ -399,6 +400,16 @@ func run(ctx *cli.Context) {
 
 	ethereum, err := eth.New(cfg)
 	if err != nil {
+		var ok bool
+		for _, no := range datadirInUseErrNos {
+			if uint(err.(syscall.Errno)) == no {
```

Comment:
This still doesn't work. 

`ethdb.NewLDBDatabase` will return a `syscall.Errno` error for which you can check. In backend.go these errors are converted to an `*errors.errorString` instance. 

```
chainDb, err := newdb(filepath.Join(config.DataDir, "chaindata"))
if err != nil {
	return nil, fmt.Errorf("blockchain db err: %v", err)
}
```
The check in main.go for an errno wil then obviously fail.

Second problem is that the check if only added to the run subcommand. Users who will start the `console` (or any other subcommand) will still get the old error message when another instance already claimed the database datadir.